app/controllers/causes_controller.rb

```diff
@@ -5,7 +5,21 @@ class CausesController < ApplicationController
   def update
     @cause.update(cause_params)
     @fmea = @cause.failure_mode.function.fmea
-    redirect_to edit_fmea_path(@fmea, anchor: card_id(@cause))
+    # redirect to the function where the cause was added
+    redirect_to edit_fmea_path(@fmea, anchor: card_id(@cause.failure_mode.function))
+  end
+
+  def create
+    @cause = Cause.new(cause_params)
+
+    if @cause.save
+      # redirect to the function where the cause was added
+      redirect_to edit_fmea_path(@cause.failure_mode.function.fmea, anchor: card_id(@cause.failure_mode.function))
+    else
+      # redirect to fmea and show errors as a flash alert
+      flash[:alert] = @cause.errors.full_messages
+      redirect_to edit_fmea_path(@cause.failure_mode.function.fmea, anchor: card_id(@cause.failure_mode.function))
+    end
```

Comment:
- Added the create method to the cause controller. 
- Managed to make the modal code quite generic using the method `element.parent.model_name.param_key` which returns the string `failure_mode` from the object `FailureMode`
- Again redirects to the parent function anchor to provide an improved UX